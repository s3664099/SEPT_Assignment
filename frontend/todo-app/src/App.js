import React, { Component } from 'react';
import Counter from './components/counter/Counter'
import TodoApp from './components/todo/TodoApp'
//import './App.css';
//import './bootstrap.css';
import Wall from './components/wall/Wall.jsx'

class App extends Component {
  render() {
    return (
      <div className="App">
        <Wall />
      </div>
    );
  }
}




export default App;