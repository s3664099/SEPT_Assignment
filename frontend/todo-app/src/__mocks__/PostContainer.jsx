export default class PostContainer {
    constructor() {
       
    }
  
    refreshPosts() {
      console.log('Mock_PostContainer: refreshPosts was called');
    }

    editPostButton(){
        console.log('Mock_PostContainer: refreshPosts was called');
    }

    deletePostButton(postID){
        console.log('delete post' + postID)
        let username = AuthenticationService.getLoggedInUserName()
        WallDataService.deletePost(username, postID)
        .then(response=>{
          this.refreshPosts()
        })
      }
    
      addPostClicked(){
        console.log("new post button clicked")
        this.props.history.push(`/wall/-1`)
      }
  }