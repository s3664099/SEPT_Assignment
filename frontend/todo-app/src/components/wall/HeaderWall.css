/*css needs to be fixed with adjustment of frame size*/

.wallnav {
    /*overflow:hidden;*/
    display: inline-block;
    width: 100%;
    color: white;
    border-bottom: 2px solid #bcbcd3;
}

.wallLogo {
    float: left;
  font-size: 32px;
    font-weight: bold;
    margin: 5px auto;
    margin-left: 10%;
    vertical-align: baseline;
    display: inline-block;
    color:black;
}

.Greetings{
    font-size: 20px;
    text-align: right;
    margin-right: 10%;
    display: inline;
    float: right;
    color:black;
}

.profilePic{
    border: 1px solid #ddd;
    border-radius: 4px;
    padding: 5px;
    width: 80px;
    border-radius: 8px;
    padding:5px;
    vertical-align: middle;
    margin: 5px 10px;
    display: inline;
}

.search{
    text-align: center;

    display: inline-block;
    width: 500px;
}

.search input {
  width: 300px;
  height: 30px;
}

.search button {
  height: 30px;
  vertical-align: middle;
  padding-top: 7px;
  padding-bottom: 23px;
}

.search ul {
  float: left;
  position:absolute;
  background-color: white;
  margin-left: 40px;
  border: 1px solid black;
  width: 300px;
  color: black;
  list-style-type: none;
}

.search ul li {
  text-align: left;
  text-decoration: none;
}
