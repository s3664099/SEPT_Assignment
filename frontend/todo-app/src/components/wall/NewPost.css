.postButton {
  height: 50px;
  width: 200px;
  margin: 20px;
  border:0px solid #05cdff;
  -webkit-border-radius: 50px;
  -moz-border-radius: 50px;
  border-radius: 50px;
  text-decoration:none;
  display:inline-block;
  color: #FFFFFF;
  background-color: #12B8FF; background-image: -webkit-gradient(linear, left top, left bottom, from(#12B8FF), to(#0F77FF));
  background-image: -webkit-linear-gradient(top, #12B8FF, #0F77FF);
  background-image: -moz-linear-gradient(top, #12B8FF, #0F77FF);
  background-image: -ms-linear-gradient(top, #12B8FF, #0F77FF);
  background-image: -o-linear-gradient(top, #12B8FF, #0F77FF);
  background-image: linear-gradient(to bottom, #12B8FF, #0F77FF);filter:progid:DXImageTransform.Microsoft.gradient(GradientType=0,startColorstr=#12B8FF, endColorstr=#0F77FF);
}

.postButton:hover{
    cursor: pointer;
}

h3{
  display: inline;
}

.postInput {
  width: 95%;
  height: 100px;
  vertical-align: top;
  border: none;
}

.newPostArea {
  width: 60%;
  margin: 30px auto;
  padding: 10px;
  border: 1px solid black;
  border-radius: 25px;
}

.userInfo {
  text-align: left;
  margin: 10px 40px;
}
