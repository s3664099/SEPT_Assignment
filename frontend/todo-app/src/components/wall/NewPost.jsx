import React, { Component } from 'react'
import { Formik, Form, Field} from 'formik';
import WallDataService from '../../api/todo/WallDataService';
import AuthenticationService from './AuthenticationService.js'
import moment from 'moment'
import {withRouter} from 'react-router-dom'

class NewPost extends Component{

    
    constructor(props){
        super(props)
        this.state ={
            postId: this.props.match.params.postId,
            postContent:'',
            ownerId: '',
            postTime: moment(new Date()).format('DD-MM-YYYY HH:MM')
        };
        
        this.handleSubmit = this.handleSubmit.bind(this)
    }
        

    componentDidMount(){
        
        if(this.state.postId === -1){
            return
        }

        let username = AuthenticationService.getLoggedInUserName()

        WallDataService.retrievePost(username, this.state.postId)
        .then(response=>this.setState({
            postContent:response.data.message,
            ownerId:response.data.ownerId,
            postTime: moment(response.data.creationTime).format('DD-MM-YYYY HH:MM')
            
        }))


    }

    handleSubmit(event){
        let username = AuthenticationService.getLoggedInUserName()
        let id = this.state.postId;
        let post={
            postId:this.state.postId,
            ownerId: this.state.ownerId,
            message:event.postContent,
            creationTime: this.state.postTime
        }
        console.log(this.state.postId)
        if(id === -1){
            console.log("create new post")
            WallDataService.createPost(username, post)
            .then(()=>this.props.history.push('/wall'))
        }else{
            console.log("update new post")
            WallDataService.toUpdatePost(username, this.state.postId, post)
            .then(
                console.log(this.props.history),
                () => this.props.history.push('/wall')
                )
            
        }
        console.log(event)
    }


    render() {
      const {postContent, postTime} = this.state
        let username = AuthenticationService.getLoggedInUserName()
        return (
            <div>
                <h1>New Post</h1>
                <h3>{username}</h3>
                <div className="newPost">
                    <Formik
                        initialValues={{ postContent, postTime }}
                        onSubmit={this.handleSubmit}
                        enableReinitialize={true}
                    >
                        {
                            (props) => (
                                <Form>
                                    <fieldset className="form-group">
                                        <label>Post content</label>
                                        <Field  type="text" name="postContent" />
                                    </fieldset>

                                    <button className="btn btn-success" type="submit">Submit</button>
                                </Form>
                            )
                        }
                    </Formik>

                </div>
            </div>

        );
    }
}

export default withRouter(NewPost)