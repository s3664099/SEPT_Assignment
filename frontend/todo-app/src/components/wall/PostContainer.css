.postContainer{
    border: 2px solid gray;
    border-radius: 5px;
<<<<<<< HEAD
    width: 90%;
    display: inline-block;
    margin:3%;
    padding:5px;
    text-align: justify;
=======
    width: 60%;
    display: inline-block;
    margin:3%;
    padding:5px;

>>>>>>> jasmine
}

.userImage, .userName, .editButton{
    display: inline-block;
<<<<<<< HEAD
}
=======
}
>>>>>>> jasmine
