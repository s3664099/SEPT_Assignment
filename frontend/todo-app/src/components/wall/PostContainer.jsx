import React, {Component} from 'react'
//import BottomBar from './BottomBar'
import './PostContainer.css'
//import CommentBox from './CommentBox'
import WallDataService from '../../api/todo/WallDataService.js'
import AuthenticationService from './AuthenticationService.js'



class PostContainer extends Component {

    
    constructor(props){
        super(props);
        this.state ={
            error : null,
            isLoaded: false,
            posts:[]
        };
    }
    
    componentDidMount(){
       let username = AuthenticationService.getLoggedInUserName()
       WallDataService.retrieveAllVisiblePosts(username)
        .then(res=>res.json())
        .then(
            (result)=>{
                this.setState({
                   isLoaded:true,
                   posts:result
                });
            },
            (error)=>{
                this.setState({
                    isLoaded:true,
                    error
                });
            }
        )
    }

    render(){
        
        const userImage = "http://placekitten.com/300/200"
        return (
                   <div>
                   {
                       this.state.posts.map((post,index)=>(
                        <div className="postContainer">
                        <div className="userImage">
                        <img className ="profilePic" src={userImage} alt="Profile Pic"></img>
                            <div className ="timeStamp">{post.creation_Time}</div>
                        </div>
                        <div className="userName"> {/*post.UserName*/} </div>
                        <div>
                            <button className ="editButton">Edit/Del</button>
                        </div>
                        <div className="userInput">{post.message}</div>
                        <hr></hr>
                       <div>{/*<BottomBar />*/}</div>
                        
                        <div>{
                            (typeof(post.comments)=='object')? 
                            <div>
                                {
                                    post.comments.map((commentBox,indexb)=>
                                    <div>
                                        <hr></hr>
                                <div>{/*<CommentBox commentbox={commentBox}/>*/}</div>
                                    </div>)
                                }
                            </div>: null
                        }</div>
                    </div>
                    ))
                   }

                    </div>
               
        );
     }

}


export default PostContainer 