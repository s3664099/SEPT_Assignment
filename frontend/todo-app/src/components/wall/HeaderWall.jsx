import React, {Component} from 'react'
import './HeaderWall.css'
import Search from './Search.jsx'

class HeaderWall extends Component {

    /*constructor(props){
        super(props);
    }*/

    

    render(){
             return (
                 
            <header>
                <nav className="wallnav">
                    <div className="wallLogo">
                        <a className ="RMITWall" href="http://www.rmit.edu.au">RMITWall</a>
                    </div>

                    <div className="search">
                      {<Search/>}
                    </div>

                    <div className="Greetings">
                        Welcome, {this.props.name}
                        <img className ="profilePic" src="http://placekitten.com/300/200" alt="Putty Tat"></img>
                    </div>
                </nav>
            </header>
        )
    }
}


export default HeaderWall
