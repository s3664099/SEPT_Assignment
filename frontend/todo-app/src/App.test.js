import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer'
import HeaderWall from './components/wall/HeaderWall';
import WallComponent from './components/wall/WallComponent'
import PostContainer from './components/wall/PostContainer';
import NewPost from './components/wall/NewPost';

//jest.mock('./src/__mock__/WallDataService')//./WallDataService../

it('App renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('HeaderWall renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HeaderWall />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('WallComponent renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WallComponent />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('PostContainer renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PostContainer />, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('header wall snapshot compare', () => {
  const tree = renderer
    .create(<HeaderWall name={'sept'}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/*
it('trial mock object', ()=>{
  expect.assertions(0);
  let username ='one';
  return expect(WallDataService.retrievePost(username, 2)).then(post=>expect(post.userID).toEqual('2'))
})
*/

it('post container test ', () => {
  const tree = renderer
    .create(<PostContainer />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

