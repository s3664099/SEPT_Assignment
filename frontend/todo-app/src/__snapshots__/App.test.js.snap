// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header wall snapshot compare 1`] = `
<header>
  <nav
    className="wallnav"
  >
    <div
      className="wallLogo"
    >
      <a
        className="RMITWall"
        href="http://www.rmit.edu.au"
      >
        RMITWall
      </a>
    </div>
    <div
      className="search"
    >
      <form>
        <input
          onChange={[Function]}
          placeholder="Search for..."
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Search
        </button>
        <ul />
      </form>
    </div>
    <div
      className="Greetings"
    >
      Welcome, 
      sept
      <img
        alt="Putty Tat"
        className="profilePic"
        src="http://placekitten.com/300/200"
      />
    </div>
  </nav>
</header>
`;

exports[`header wall snapshot compare 2`] = `
<header>
  <nav
    className="wallnav"
  >
    <div
      className="wallLogo"
    >
      <a
        className="RMITWall"
        href="http://www.rmit.edu.au"
      >
        RMITWall
      </a>
    </div>
    <div
      className="search"
    >
      <form>
        <input
          onChange={[Function]}
          placeholder="Search for..."
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Search
        </button>
        <ul />
      </form>
    </div>
    <div
      className="Greetings"
    >
      Welcome, 
      sept
      <img
        alt="Putty Tat"
        className="profilePic"
        src="http://placekitten.com/300/200"
      />
    </div>
  </nav>
</header>
`;

exports[`new post test  1`] = `
<div>
  <div>
    <button
      className="newPostButton"
      onClick={[Function]}
    >
       Create New Post :)
    </button>
  </div>
</div>
`;

exports[`post container test  1`] = `
<div>
  <div>
    <button
      className="newPostButton"
      onClick={[Function]}
    >
       Create New Post :)
    </button>
  </div>
</div>
`;
